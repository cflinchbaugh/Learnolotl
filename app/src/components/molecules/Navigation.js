import React, { Component } from 'react';
import { Link } from 'react-router';

class Navigation extends Component {
    render() {
        return (
            <div>
                <Link to="/">Home</Link>
                <Link to="/test">Deck Test</Link>
            </div>
        )
    }
}

export default Navigation