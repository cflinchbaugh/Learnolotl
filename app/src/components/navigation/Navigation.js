import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';


const StyleWrapper = styled.div`
    width: 100%;
    background-color: ${props => props.theme.primary};
    margin-bottom: 30px;

`
    
const NavLinkStyled = styled(NavLink)`
    background-color: ${props => props.theme.primary};
    color: ${props => props.theme.primaryColor};
    text-decoration: none;
    transition: background-color 0.25s;
    padding: 10px 8px;
    display: inline-block;

    &:hover {
        background-color: ${props => props.theme.primaryDarkMedium};
        color: ${props => props.theme.primaryDarkMediumColor};
    }

    &.${(props) => props.activeClassName } {
        background-color: ${props => props.theme.primaryDark};
    }

`
NavLinkStyled.defaultProps = {
    activeClassName: 'active'
};

function Navigation () {
    return (
        <StyleWrapper>
            <div className="navigation-wrapper">
                <NavLinkStyled to="/learn">Learn</NavLinkStyled>
                <NavLinkStyled to="/build">Build</NavLinkStyled>
                <NavLinkStyled to="/about">About</NavLinkStyled>
            </div>
        </StyleWrapper>
    )
}

export default Navigation