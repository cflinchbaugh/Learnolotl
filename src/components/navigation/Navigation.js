import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';


const StyleWrapper = styled.div`
    width: 100%;
    margin-bottom: 30px;

`
    
const NavLinkStyled = styled(NavLink)`
    color: ${props => props.theme.primaryColor};
    text-decoration: none;
    transition: color 0.25s;
    margin: 10px 8px;
    display: inline-block;

    &:hover {
        color: #47517c;
    }

    &.${(props) => props.activeClassName } {
        color: #3c466b;
    }

`
NavLinkStyled.defaultProps = {
    activeClassName: 'active'
};

function Navigation(props) {
    let navigationItems = props.navigationItems.map((itemData, idx) => {
        return (
            <NavLinkStyled {...itemData} key={idx}>
                {itemData.label}
            </NavLinkStyled> 
        )
    })
    return (
        <StyleWrapper>
            <div className="navigation-wrapper">
                {navigationItems}
            </div>
        </StyleWrapper>
    )
}

export default Navigation