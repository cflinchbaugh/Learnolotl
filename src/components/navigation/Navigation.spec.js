import React from 'react';
import {shallow} from 'enzyme';
import Navigation from './Navigation';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Navigation', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const navigationData = {

        }

        const tree = renderer.render(
            <Navigation {...navigationData}/>
        );

        expect(tree).toMatchSnapshot();
    });

    
});