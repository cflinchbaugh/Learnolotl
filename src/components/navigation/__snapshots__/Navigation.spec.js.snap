// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders and matches snapshot 1`] = `
<styled.div>
  <div
    className="navigation-wrapper"
  >
    
  </div>
</styled.div>
`;
